yb-voyager/cmd/migration_complexity.go

```diff
@@ -200,3 +210,121 @@ func getComplexityForLevel(level string, count int) string {
+</table>
+
+<p>
+	Level-1 Impact: Resolutions are available with minimal effort.<br/>
+	Level-2 Impact: Resolutions are available requiring moderate effort.<br/>
+	Level-3 Impact: Resolutions may not be available or are highly complex.
+</p>
+
+<p>
+	<strong>Reasoning:</strong> {{ .ComplexityRationale }}
+</p>
+`
+
+const explainTemplateText = `Reasoning: {{ .ComplexityRationale }}`
+
+type ExplanationData struct {
+	Summaries           []CategorySummary
+	ComplexityRationale string // short reasoning or explanation text
+}
+
+type CategorySummary struct {
+	Category        string
+	TotalIssueCount int
+	ImpactCounts    map[string]int // e.g. {"Level-1": 3, "Level-2": 5, "Level-3": 2}
+}
+
+func buildMigrationComplexityExplanation(sourceDBType string, assessmentReport AssessmentReport, reportFormat string) (string, error) {
+	if sourceDBType != POSTGRESQL {
+		return "", nil
+	}
+
+	var explanation ExplanationData
+	explanation.ComplexityRationale = migrationComplexityRationale
+
+	explanation.Summaries = buildCategorySummary(assessmentReport.Issues)
+
+	var tmpl *template.Template
+	var err error
+	if reportFormat == "html" {
+		tmpl, err = template.New("Explain").Parse(explainTemplateHTML)
+	} else {
+		tmpl, err = template.New("Explain").Parse(explainTemplateText)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("failed creating the explanation template: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, explanation); err != nil {
+		return "", fmt.Errorf("failed executing the template with data: %w", err)
+	}
+	return buf.String(), nil
+}
+
+func buildRationale(finalComplexity string, l1Count int, l2Count int, l3Count int) string {
+	switch finalComplexity {
+	case constants.MIGRATION_COMPLEXITY_HIGH:
+		return fmt.Sprintf("Found %d Level-2 issue(s) and %d Level-3 issue(s), resulting in HIGH migration complexity", l2Count, l3Count)
+	case constants.MIGRATION_COMPLEXITY_MEDIUM:
+		return fmt.Sprintf("Found %d Level-1 issue(s), %d Level-2 issue(s) and %d Level-3 issue(s), resulting in MEDIUM migration complexity", l1Count, l2Count, l3Count)
+	case constants.MIGRATION_COMPLEXITY_LOW:
+		return fmt.Sprintf("Found %d Level-1 issue(s) and %d Level-2 issue(s), resulting in LOW migration complexity", l1Count, l2Count)
+	}
+	return ""
+}
+
+func buildCategorySummary(issues []AssessmentIssue) []CategorySummary {
+	summaryMap := make(map[string]*CategorySummary)
+	for _, issue := range issues {
+		if issue.Category == "" {
+			continue // skipping unknown category issues
+		}
+
+		if _, ok := summaryMap[issue.Category]; !ok {
+			summaryMap[issue.Category] = &CategorySummary{
+				Category:        issue.Category,
+				TotalIssueCount: 0,
+				ImpactCounts:    make(map[string]int),
+			}
+		}
+
+		summaryMap[issue.Category].TotalIssueCount++
+		summaryMap[issue.Category].ImpactCounts[issue.Impact]++
+	}
+
+	var result []CategorySummary
+	for _, summary := range summaryMap {
+		summary.Category = utils.ToTitleCase(summary.Category)
```

Comment:
In golang layer, it would be good to still retain the default category values. This is a UI requirement, so it makes sense to do it at the HTML layer. You can export the `ToTitleCase` func to the template, and call it.